src/natcap/invest/coastal_blue_carbon/coastal_blue_carbon.py

```diff
@@ -188,941 +187,1485 @@
+            Required if ``args['do_economic_analysis']``.
+        args['biophysical_table_path'] (string): The path to the biophysical
+            table on disk.  This table has many required columns.  See
+            ``ARGS_SPEC`` for the required columns.
+        args['landcover_transitions_table'] (string): The path to the landcover
+            transitions table, indicating the behavior of carbon when the
+            landscape undergoes a transition.
 
     Returns:
-        transition_idx (int): the current transition
+        ``None``.
+
     """
-    for i in range(0, transitions):
-        if timestep < (snapshot_years[i+1] - snapshot_years[0]):
-            return i
+    suffix = utils.make_suffix_string(args, 'results_suffix')
+    output_dir = os.path.join(args['workspace_dir'], 'output')
+    intermediate_dir = os.path.join(args['workspace_dir'], 'intermediate')
+    taskgraph_cache_dir = os.path.join(intermediate_dir, 'task_cache')
 
+    utils.make_directories([output_dir, intermediate_dir, taskgraph_cache_dir])
 
-def s_to_timestep(snapshot_years, snapshot_idx):
-    """Convert snapshot index position to timestep.
+    try:
+        n_workers = int(args['n_workers'])
+    except (KeyError, ValueError, TypeError):
+        # KeyError when n_workers is not present in args
+        # ValueError when n_workers is an empty string.
+        # TypeError when n_workers is None.
+        n_workers = -1  # Synchronous mode.
+    task_graph = taskgraph.TaskGraph(
+        taskgraph_cache_dir, n_workers, reporting_interval=5.0)
+
+    snapshots = _extract_snapshots_from_table(args['landcover_snapshot_csv'])
+
+    # Phase 1: alignment and preparation of inputs
+    baseline_lulc_year = min(snapshots.keys())
+    baseline_lulc_path = snapshots[baseline_lulc_year]
+    baseline_lulc_info = pygeoprocessing.get_raster_info(
+        baseline_lulc_path)
+    min_pixel_size = numpy.min(numpy.abs(baseline_lulc_info['pixel_size']))
+    target_pixel_size = (min_pixel_size, -min_pixel_size)
+
+    transition_years = set()
+    analysis_year = int(args['analysis_year'])
+
+    base_paths = [baseline_lulc_path]
+    aligned_lulc_paths = {}
+    aligned_paths = [os.path.join(
+        intermediate_dir,
+        f'aligned_lulc_baseline_{baseline_lulc_year}{suffix}.tif')]
+    aligned_lulc_paths[baseline_lulc_year] = aligned_paths[0]
+    for snapshot_year in snapshots:
+        if snapshot_year == baseline_lulc_year:
+            continue
+        base_paths.append(snapshots[snapshot_year])
+        transition_years.add(snapshot_year)
+        aligned_paths.append(
+            os.path.join(
+                intermediate_dir,
+                f'aligned_lulc_snapshot_{snapshot_year}{suffix}.tif'))
+        aligned_lulc_paths[snapshot_year] = aligned_paths[-1]
+
+    prices = None
+    if ('do_economic_analysis' in args and
+            args['do_economic_analysis'] not in ('', None)):
```

Comment:
The UI will always include the value of `do_economic_analysis`, but the same can't be said about someone using the model through a script.

You're very right about this condition being incorrect!  I've patched it and updated it to use the `.get()` notation above (and also further simplifying because we can rely on python's truthiness here) in 26f8b326